source/Ship.cpp

```diff
@@ -3899,6 +3914,33 @@ void Ship::SetFormationPattern(const FormationPattern *formationToSet)
 
 
 
+const vector<Ship *> &Ship::GetShipsTargetingThis() const
+{
+	return targetingList;
+}
+
+
+
+const double Ship::GetTargeterStrength() const
+{
+	return targeterStrength;
+}
+
+
+
+double Ship::UpdateTargeterStrength()
+{
+	targeterStrength = targeterStrength < 1. ? 0 : targeterStrength / 1.05;
+
+	for(Ship *targeter : targetingList)
+		if(targeter != nullptr)
+			targeterStrength += targeter->Strength();
+
+	return targeterStrength;
+}
```

Comment:
You're not using the return value in AI. Might as well not even return it.
Also, you should leave a comment explaining why targeterStrength decays. (Ziproot asked for this but it never happened.)
```suggestion
void Ship::UpdateTargeterStrength()
{
	targeterStrength = targeterStrength < 1. ? 0 : targeterStrength / 1.05;

	for(Ship *targeter : targetingList)
		if(targeter != nullptr)
			targeterStrength += targeter->Strength();
}
```